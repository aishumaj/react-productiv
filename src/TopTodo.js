import React from "react";

import Todo from "./Todo";

/** Shows the top todo.
 *
 * Props:
 * - todos = [{id, title, description, priority}... ]
 *
 * TodoApp -> TopTodo
 */

function TopTodo({ todos }) {
  // lowest-priority # is the highest priority
  let {id, title, description, priority} = todos.reduce(
    (acc, cur) => cur.priority < acc.priority ? cur : acc, todos[0]);

  return (
  <Todo id={id} title={title} description={description} priority={priority}/>
  )}

export default TopTodo;