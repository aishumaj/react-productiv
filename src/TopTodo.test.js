import React from "react";
import { render } from "@testing-library/react";
import TopTodo from "./TopTodo"

const testTodos = [{id:1, title:"hello", description: "test", priority:2}, 
  {id:1, title:"top priority", description: "test2", priority:1}, 
  {id:1, title:"hello", description: "test3", priority:3}]

describe("productiv app", function () {
  it("renders without crashing", function () {
    render(<TopTodo todos={testTodos} />);
  });

  it("contains expected title and description", function () {
    const result = render(<TopTodo todos={testTodos}/>);
    expect(result.queryByText("top priority")).toBeInTheDocument();
    expect(result.queryByText("test2")).toBeInTheDocument();
  });
  
  test("matches snapshot", function () {
    const { container } = render(<TopTodo todos={testTodos}/>);
    expect(container).toMatchSnapshot();
  });
  
});