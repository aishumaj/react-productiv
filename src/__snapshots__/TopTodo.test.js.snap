// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`productiv app matches snapshot 1`] = `
<div>
  <div
    class="Todo"
    id="1"
  >
    <div>
      <b>
         
        top priority
         
      </b>
       
      <small>
         (priority: 
        1
        )
      </small>
    </div>
    <div>
      <small>
        test2
      </small>
    </div>
  </div>
</div>
`;
